search_engine_parser/core/engines/baidu.py

```diff
@@ -16,15 +17,32 @@ class BaiduSearch(BaseSearch):
 	" companies in the world.\n\tBaidu offers various services, including a"\
 	" Chinese search engine, as well as a mapping service called Baidu Maps."
 
+	
+	"""Override get_search_url"""
+	def get_search_url(self, query=None, page=None):
+		""" 
+		Return a formatted search url.
+		Offsets are of form 0,10,20, etc. So if 1 is passed, we make it 0, for 2->(2-1)*10=10. etc.
+		Adding offset value to self as we would need it in parse_soup
+		"""
+		
+		self.offset = (page - 1) * 10
+		return  self.search_url.format(query=query, page=page, offset=self.offset)
+	
+	
 	def parse_soup(self, soup):
 		"""
 		Parses Baidu for a search query
 		"""
 		
-		"""Baidu search can be made deterministic via an id, so add it in self"""
+		"""Baidu search can be made deterministic via an id.
+		The id of divs are in range 1-10 for page 1, 11-20 in page 2, so that arithmetic is
+		handled by the previous calculation of offset (it transforms 1 to 0, 2 to 10, etc.)
+		and we loop for all div's from (10+1==11) to (10+11-1)=20 {-1 due to nature of range function}
+		"""
 		
 		soups=[]
-		for i in range(int(self.offset)*10+1,int(self.offset)*10+11):
+		for i in range(int(self.offset) + 1,int(self.offset) + 11):
```

Comment:
How about using `re` to compile and search for the divs instead. https://www.crummy.com/software/BeautifulSoup/bs4/doc/#a-regular-expression

Reply:
Actually, that doesn't seem feasible. 
A quick example: If you search the text **preaching** on Baidu, the top div with all data has a class called *result-op c-container xpath-log*, the 2nd div has class *result c-container * while the 3rd has *result-op c-container*. While c-container class is common in these 3, I received less results in many cases, which lead me to thinking that using **id's** is the best possible way.

However, now that I have taken another look, it seems that only *div's* with relevant results do contain numeric ids, so we can go with something like:

```[python]
import re
return re.findall('<div.*?id=[0-9]{1,2}.*?>',soup))
```
Is this what you want for a cleaner API (that is, not to add the offset parameter to self)? 🤔

Reply:
Used regex as per the idea.

Reply:
Reverted base.py as instructed 😄